### ISSUE TITLE
psconvert sometimes fails to convert PS to PDF

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#300]
**Description of the problem**
[GMT_linearrow.sh](https://github.com/GenericMappingTools/gmt/blob/master/doc_classic/scripts/GMT_linearrow.sh) and [GMT_lineoffset.sh](https://github.com/GenericMappingTools/gmt/blob/master/doc_classic/scripts/GMT_lineoffset.sh) are two scripts used in the GMT cookbook.

These two scripts can generate PS images correctly. However, PDF files generated by `gmt psconvert -A -P -Tf xxx.ps` seem damaged, but other formats (e.g. PNG, JPG) are OK.

**Full script that generated the error**

First run the scripts, then use `gmt psconvert -A -P -Tf xxx.ps` to generate the PDF files.

**Full error message**

![image](https://user-images.githubusercontent.com/3974108/50877223-ee94cb00-139e-11e9-8497-219acfb23659.png)

**System information**

* Operating system: macOS
* Version of GMT: 6.0.0_26fa34c
* Version of gs: 9.26

BTW: No such issue for GMT 5.4.5.

**Comment by PaulWessel** [comment]
Are you able to diff the 6.0.0 and 5.4.5 ps files? I tried Adobe Distiller and it was able to convert to PDF.  But psconvert's output does not open in Acrobat.

**Comment by joa-quim** [comment]
Found the culprit.

**Comment by joa-quim** [comment]
Waste no time. The culprit is the GeoPDF block that gets added by GMT6 psconvert but not by GMT5.
Had no time to investigate this any further.

**Comment by joa-quim** [comment]
To make it more clear. GMT produces GeoPDFs (just confirmed that for example converting GMT_lambert_conic.ps gives a correct GeoPDF), but for some reason some of these GeoPDFs, like the cases in this issue, produce invalid PDF.

**Comment by joa-quim** [comment]
Damn it. This is enough to show the issue
```
psbasemap  -R8/32/40/44 -JM12 -Baf -P > lixo.ps
psconvert -A -Tf lixo.ps
`` 
It seems like not all maps can be GeoPDFs. But we don't want to drop the GeoPDF facility (that was an issue in SVN brought by @KristofKoch).

Alternatives? Make GeoPDFs only with a psconvert option?

**Comment by PaulWessel** [comment]
Perhaps place the PS code that does this under an if test and then psconvert could pass that variable as true if an option or flag is set.

**Comment by KristofKoch** [comment]
@joa-quim I'm very thankful that you implemented and refined the GeoPDF functionality for me - however im still struggling mastering the art of creating them. There seems to be a problem with page orientation and format/size of the plot. I'm not yet able to reliably generate valid GeoPDFs or reliably generate invalid ones. However this might be a GhostScript (9.26 on the dev-machine) problem.

**Comment by KristofKoch** [comment]
> There seems to be a problem with page orientation and format/size of the plot. I'm not yet able to reliably generate valid GeoPDFs or reliably generate invalid ones.

Upon some further investigation it seem the media must be in portrait mode `-P` and the plot must be smaller than the width of the media to properly function.

**Comment by joa-quim** [comment]
But my example above fulfills those conditions but still the PDF is corrupted.

**Comment by joa-quim** [comment]
@KristofKoch you should see #302 as it now requires **-Qp** to create GeoPDFs

**Comment by seisman** [comment]
Fixed in #302.

**Comment by KristofKoch** [comment]
@joa-quim thank you for the heads up!
